### PR TITLE
DeviceTypeForm refactor

### LINKED ISSUE #3414: Update `Create new device type` to include test length
### Description
`Create new device type` screen doesnt include test length
current screen:
image

### Proposed solution 
add test length to screen and improve general layout, follow the following for the general layout, not the style.

CREATE_DEVICE_TYPE

### PR DESCRIPTION
**Pull request by mehansen**
# FRONTEND PULL REQUEST

## Related Issue

- Resolves #3414 

## Changes Proposed

frontend:
- consolidate ManageDevicesForm and DeviceTypeForm into one component DeviceForm, which conditionally shows a select device dropdown
- Dan's changes to ManageDevicesForm to add the test length and change the formatting now also apply to the the Create new device type form as well, since they're using the same component
- added auto, inc an additional snapshot for DeviceTypeFormContainer

backend:
- support creating devices with test length parameter
- remove backend code that assumes testLength based on name

## Additional Information

- I feel like more of the duplicate code (e.g. swabOptions, diseaseOptions) could've been moved from the two FormContainers into DeviceForm, but I felt like this change was getting big and was also unsure why we did it in the form container in the first place - is it best practice to query and then pass results into a sub-component?
- known already-existing issues I saw, specifically with the multiselects:
  - if a swab or disease option is already selected for a device in the manage devices form, it'll still show up in the dropdown, whereas once you select a new option that one will be removed from the dropdown
  - if a device has a swab or disease option specified, the "disable save button until form changed" functionality does not work - we consider the form changed when populating the swab or disease option on device load

## Testing

go check it out in test :) or run it locally
key functionality:
- editing existing device and creating new device save properly
- new test length field for create
- alert and no save on create form if invalid test length (val <= 0 || val > 999)
 
## Screenshots / Demos

https://user-images.githubusercontent.com/6401323/184455406-b9d496c5-18c4-4906-bee1-5a304823d1fa.mov

### Design
- [ ] Any UI/UX changes have a designer as a reviewer, and changes have been approved
- [ ] Any large-scale changes have been deployed to `test`, `dev`, or `pentest` and smoke-tested by both the engineering and design teams

### Content
- [n/a] Any content changes have been approved by content team

### Support
- [n/a] Any changes that might generate new support requests have been flagged to the support team
- [n/a] Any changes to support infrastructure have been demo'd to support team

### Security
- [n/a] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [n/a] Any dependencies introduced have been vetted and discussed

### Documentation
- [n/a] Any changes to the startup configuration have been documented in the README

### CODE REVIEW

File: frontend/src/app/supportAdmin/DeviceType/ManageDeviceTypeFormContainer.tsx [hunk 1]
```diff
@@ -28,21 +29,38 @@ const ManageDeviceTypeFormContainer = () => {
...
+            body="The device has been updated"
+          />
+        );
+        showNotification(alert);
+        setSubmitted(true);
+      });
+    } else {
+      //todo: make this better
+      console.log(
+        "internal id for saving a device is undefined and your code is bad; aborting"
```
Comment (nathancrtr) [inline frontend/src/app/supportAdmin/DeviceType/ManageDeviceTypeFormContainer.tsx:61]: 😂

File: frontend/src/app/supportAdmin/DeviceType/DeviceForm.test.tsx [hunk 2]
```diff
@@ -1,24 +1,91 @@
 import { render, screen, within } from "@testing-library/react";
 import userEvent from "@testing-library/user-event";
 
-import ManageDevicesForm from "./ManageDevicesForm";
-import { addValue } from "./DeviceTypeForm.test";
```
Comment (mehansen) [inline frontend/src/app/supportAdmin/DeviceType/DeviceForm.test.tsx:5]: importing a method from FileA.test.tx into FileB.test.tx seemed to cause running FileB to trigger running all tests in FileA as well, which was throwing off my snapshots, so now all 3 frontend test files have their own `addValue` method. curious if anyone knows what that's about or if I should make a testutils or something

**Comment by mehansen** [comment]
@vz3 pretty minor, but I accidentally added bonus functionality of defaulting test length to 15 (see demo recording) when opening the create device form. would it be preferable to have it initialize blank or 0?

**Review by zdeveloper — DISMISSED** [review]
LGTM! thank you consolidating into one component!

File: frontend/src/app/supportAdmin/DeviceType/DeviceTypeFormContainer.tsx [hunk 3]
```diff
@@ -32,20 +23,36 @@ const DeviceTypeFormContainer = () => {
...
+          fetchPolicy: "no-cache",
+        }).then(() => {
+          const alert = (
+            <Alert
+              type="success"
+              title="Created Device"
+              body="The device has been created"
+            />
+          );
+          showNotification(alert);
```
[the remark this answers is not in the document]
Comment (mehansen) [reply frontend/src/app/supportAdmin/DeviceType/DeviceTypeFormContainer.tsx:47]: hmmm that's weird I remember seeing the alert before. this is happening in main too. looking into if this impacts all our alerts or if it's just device form related

**Review by DanielSass — APPROVED** [review]

**Comment by vz3** [comment]
On the 15m default I'm good with that bc I believe that's the most common test time.

**Review by zdeveloper — APPROVED** [review]
LGTM!

**Pull request merged**